docs/sdks/python/knowledge-bases.mdx

```diff
@@ -0,0 +1,1069 @@
+---
+title: Knowledge Base
+sidebarTitle: Knowledge Bases
+---
+
+A knowledge base is an advanced system that organizes information based on semantic meaning rather than simple keyword matching. It integrates embedding models, reranking models, and vector stores to enable context-aware data retrieval.
+
+By performing semantic reasoning across multiple data points, a knowledge base delivers deeper insights and more accurate responses, making it a powerful tool for intelligent data access.
+
+## How Knowledge Bases Work
+
+Before diving into the syntax, here is a quick walkthrough showing how knowledge bases work in MindsDB.
+
+We start by creating a knowledge base and inserting data. Next we can run semantic search queries with metadata filtering.
+
+<Steps>
+  <Step title="Create a knowledge base">
+    Use the `create()` function to create a knowledge base, specifying all its components.
+
+    ```python
+    server = mindsdb_sdk.connect()
+    project = server.get_project()
+
+    my_kb = project.knowledge_bases.create(
+        'my_kb',
+        embedding_model={'provider': 'openai', 'model_name': 'text-embedding-3-small', 'api_key': 'sk-...'},
```

Comment:
Should we explain all parameters e.g `storage` looks confusing like what user needs to provide?